applications/iris_followup.md

```diff
@@ -258,109 +210,160 @@ Brian believes that data-driven success is something that often requires top lev
+
+2. As part of this milestone, we formally build our testnet, which will include the development of CICD pipelines for automated runtime upgrades.
 
 | Number | Deliverable | Specification |
 | -----: | ----------- | ------------- |
 | 0a. | License | Apache 2.0 |
 | 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
 | 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. We will provide a demo video and a manual testing guide, including environment setup instructions. |
 | 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
-| 1. | Substrate Module: Iris-Assets/Iris-Session | Implement zk proofs that are submitted by offchain workers after communicating with IPFS (in the Iris-Session pallet) and verify them in the Iris-Assets pallet |
-| 2. | Substrate Module | Implement a native token, IRIS, and a secondary token, OBOL, and a function that allows nodes to stake IRIS and be rewarded with OBOL based on the IRIS-OBOL conversion rate |
-| 3. | Substrate Module: Iris-Storage | Implement generation of dummy files of specific sizes, storage of these files by storage node candidates, and submission of a zk proof after submission. |
-| 4. | Substrate Module: Iris-Storage | Implement Storage Node routing module to match storage requests with nodes who meet minimal requirements as defined by data owners |
-| 5. | Substrate Module: Iris-Storage | Implement  the T-MAN protocol and the relative scoring function |
-| 6. | Substrate Module: Iris-Storage | Implement the stochastic replication game to ensure availability and replication of data |
-| 7. | User Interface | We modify the user interface to allow data owners to rent storage space within the network, as well as a simple view for storage nodes to view rewards that they have accrued. |
+| 0e. | Article | We will publish a medium article explaining what was achieved as part of the grant, as well as any additional learnings that we deem important throughout development. https://medium.com/ideal-labs |
+| 2. | Substrate Module: Generic Storage Service pallet | We build a generic pallet with read and write capabilities which can be modified to support multiple storage systems. |
+| 3. | Substrate Module: Centralized Storage System | We build a storage system connector based on (2) which can read and write data to a centralized storage system (i.e. an AWS S3 or equivalent local file server). |
+| 4. | Substrate Module: Integration with Crust | We use the pallet developed during part 2 to use XCM to store and retrieve data in the Crust network. |
```

Comment:
You're absolutely right, I did gloss over some of the details. I've updated the storage system integration section to include more information on our intended approach. 

To give an overview: we plan on using two pallets to enable the integration with Crust. Firstly the [xStorage](https://github.com/crustio/crust/blob/parachain/shadow/crust-collator/pallets/xstorage/src/lib.rs) pallet developed by the crust team to allow us to submit cross chain storage orders. In conjunction with the xStorage pallet, we will integrate with the [xTokens](https://github.com/open-web3-stack/open-runtime-module-library/tree/master/xtokens) pallet so that we can allow for users to pay for storage fees in our native token (IRIS). Our storage orders will specify the CID and multiaddress of the proxy node's IPFS node from which Crust's IPFS nodes should fetch data. I've also reached out to the Crust team to start opening a dialogue with them so that we can efficiently perform this integration.

We intend to abstract this from the user by creating a 'generic storage connector pallet' which acts as a generic interface for storage systems and would behave the same regardless of the underlying storage system. In the future, if our dev team and presence grows, we do still want to introduce our own storage system for a more integrated solution, without its introduction being disruptive for the user. Specifically, the pallet will expose extrinsics to data owners and consumers that allow them to 'read' and 'write' data, which ultimately results in commands dispatched to proxy nodes which are aware of the underlying implementation.

Reply:
I'm not completely sure how the flow of that will work exactly. I've reached out to start a dialogue with devs from Crust but haven't made any headway quite yet. According to their docs, both Acala and Darwinia have implemented this style of communication with Crust, so I'm looking to those networks for some inspiration/guidance on this as well.

Reply:
Yes, I would have liked to see their integration or documentation on it but couldn't find anything. I'd remain cautious about considering this fully functional, but it should be interesting for you to take this approach as far as possible so I'm happy to approve the amendment. Feel free to amend again later on.